_source/_posts/2022-05-04-angular-microfrontend-deploy.md

```diff
@@ -0,0 +1,286 @@
+---
+layout: blog_post
+title: "Secure and Deploy Micro-Frontends in Angular"
+author: alisa-duncan
+by: advocate
+communities: [javascript]
+description: "A tutorial to prepare your micro-frontend project for production by using dynamic module loading, then configure deployment using Vercel. "
+tags: [angular, typescript]
+tweets:
+- ""
+- ""
+image: 
+type: conversion
+github: https://github.com/oktadev/okta-angular-microfrontend-example
+---
+
+Micro-frontends continue to gain a lot of interest and traction in front-end development. The architecture models the same concept as micro-services - as a way to decompose monolithic front-end applications. And just like with micro-services, micro-frontends have complexities to manage.
+
+This post is part two in a series where we'll build an e-commerce site using Angular using micro-frontends. We're using Webpack 5 with Module Federation to support wiring the micro-frontends together, demonstrate sharing authenticated state between the different front-ends, prepare for deployment using dynamic module loading, and deploy it all to Vercel's free hosting plan. 
+
+In this second post in the series, we're building on the site we created in the first post. Check out [How to Build Micro-Frontends Using Module Federation in Angular](/blog/2022/05/02/angular-microfrontend-auth). Let's continue working on our cupcake e-commerce site by adding dynamic module loading, securing unprotected routes, and setting up deploying the site in Vercel. 🎉
```

Comment:
```suggestion
In this second post, we're building on the site we created in part one, [How to Build Micro-Frontends Using Module Federation in Angular](/blog/2022/05/02/angular-microfrontend-auth). Let's add dynamic module loading to our cupcake e-commerce site, secure unprotected routes, and set up the site for deployment in Vercel, a service designed to improve the front-end development workflow. 🎉
```
streamlining, and added a very brief description of Vercel, which might not be familiar to all readers. I thought about adding a link - your call.